supervisor/backups/backup.py

```diff
@@ -340,7 +340,9 @@ def _create_backup():
         finally:
             self._tmp.cleanup()
 
-    async def store_addons(self, addon_list: list[str]) -> list[Awaitable[None]]:
+    async def store_addons(
+        self, addon_list: list[str], *, on_complete: Callable[[], None] | None = None
```

Comment:
Instead having an `on_complete` callback, can we not create jobs out of `_addon_save` and `_save` and `_restore` (could be still inline function) - I guess we need create an `job_inline` helper decorator as the `job` is more top-level. Then we can add key for step directly as parameter to `job` and `job_inline`. I believe it would be possible to handle this job tracking complete inside the job handler/declerator without modify any other parts of the code, but of course, we need declare more function with the job decorator as now we have an use case for doing this

Reply:
maybe to track the state (or what is ongoing), we even not need a new key as parameter, I guess we can use the `name` as step